lib/msf/core/post/windows/mssql.rb

```diff
@@ -109,13 +107,28 @@ def check_sqlcmd
         # @param [String] query the query to execute
         # @param [String] instance the SQL instance to target
         # @param [String] server the SQL server to target
+        # @param [String] database the database to connect to upon successfully authenticating
+        # @param [String] username the username to authenticate as
+        # @param [String] password the password to authenticate with
         # @return [String] the result of query
-        def run_sql(query, instance = nil, server = '.')
+        def run_sql(query, instance = nil, server = '.', database: nil, username: nil, password: nil)
           target = server
           if instance && instance.downcase != 'mssqlserver'
             target = "#{server}\\#{instance}"
           end
-          cmd = "#{@sql_client} -E -S #{target} -Q \"#{query}\" -h-1 -w 200"
+          cmd = sql_client.to_s
+          cmd += " -d #{database}" if database.present?
+          if username.present? && password.nil?
+            raise ArgumentError, 'Username provided but no value for the password was provided!'
+          elsif username.present? && !password.nil?
+            cmd += " -U \"#{username}\" -P \"#{password}\""
+          elsif username.blank? && password.present?
```

Comment:
This fails in the case where the current user has a blank password.

```suggestion
          elsif username.blank? && !password.nil?
```

Reply:
Good spotting, thanks for reporting this! Will fix this now.